source/loader/layers/sanitizer/symbolizer_llvm.cpp

```diff
@@ -0,0 +1,66 @@
+/*
+ *
+ * Copyright (C) 2024 Intel Corporation
+ *
+ * Part of the Unified-Runtime Project, under the Apache License v2.0 with LLVM Exceptions.
+ * See LICENSE.TXT
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ *
+ * @file symbolizer_llvm.cpp
+ *
+ */
+
+#include "symbolizer_llvm.hpp"
+#include "ur_sanitizer_layer.hpp"
+
+#include <sstream>
+#include <string>
+
+namespace ur_sanitizer_layer {
+
+std::vector<std::unique_ptr<SymbolizerTool>> SymbolizerTools;
+
+namespace {
+
+bool ExtractSourceInfo(const std::string &output, SourceInfo &SI) {
+    auto p1 = output.find('\n');
+    std::string function = output.substr(0, p1);
+    auto p2 = output.find(':', ++p1);
+    std::string file = output.substr(p1, p2 - p1);
+    auto p3 = output.find(':', ++p2);
+    int line = std::stoi(output.substr(p2, p3 - p2));
+    int column = std::stoi(output.substr(++p3));
+    if (function != "??") {
+        SI.function = std::move(function);
+    }
+    if (file != "??") {
+        SI.file = std::move(file);
+    }
+    SI.line = line;
+    SI.column = column;
+    return true;
+}
+
+} // namespace
+
+bool LLVMSymbolizer::symbolizePC(const BacktraceInfo &BI, SourceInfo &SI) {
+    std::stringstream ss;
+    ss << "llvm-symbolizer --obj=" << BI.module << " " << BI.offset;
```

Comment:
would it be possible to craft such BI.module to make the process execute something unexpected?

Reply:
> Do we absolutely need to run a binary this way? There's no way around it? How is cmd sanitized?
> would it be possible to craft such BI.module to make the process execute something unexpected?

Just from produce of our code, "BI.module" is get from 

https://github.com/oneapi-src/unified-runtime/blob/4eefa5be4688c960735c12fc844e5f37ec6704e3/source/loader/layers/sanitizer/linux/backtrace.cpp#L33-L34

So it depends on the symbol information of binary by "backtrace_symbols".

Yes, I think it has the risk of being injected by other commands. 
I'll find a more suitable way to use "llvm-symbolizer". 
Currently, can I submit an issue to track this problem?

Reply:
@wlemkows what do you think?

For me it's ok, but you'd need to absolutely make sure this is addressed before this code is released.